admin/code/CampaignAdmin.php

```diff
@@ -0,0 +1,408 @@
+	"id": "EditForm",
+	"schema": {
+		"name": "EditForm",
+		"id": "EditForm",
+		"action": "schema",
+		"method": "GET",
+		"schema_url": "admin\/campaigns\/schema\/EditForm",
+		"attributes": {
+			"id": "Form_EditForm",
+			"action": "admin\/campaigns\/EditForm",
+			"method": "POST",
+			"enctype": "multipart\/form-data",
+			"target": null,
+			"class": "cms-edit-form CampaignAdmin LeftAndMain"
+		},
+		"data": [],
+		"fields": [{
+			"name": "ID",
+			"id": "Form_EditForm_ID",
+			"type": "Hidden",
+			"component": null,
+			"holder_id": null,
+			"title": false,
+			"source": null,
+			"extraClass": "hidden nolabel",
+			"description": null,
+			"rightTitle": null,
+			"leftTitle": null,
+			"readOnly": false,
+			"disabled": false,
+			"customValidationMessage": "",
+			"attributes": [],
+			"data": []
+		}, {
+			"name": "ChangeSets",
+			"id": "Form_EditForm_ChangeSets",
+			"type": "Custom",
+			"component": "GridField",
+			"holder_id": null,
+			"title": "Campaigns",
+			"source": null,
+			"extraClass": null,
+			"description": null,
+			"rightTitle": null,
+			"leftTitle": null,
+			"readOnly": false,
+			"disabled": false,
+			"customValidationMessage": "",
+			"attributes": [],
+			"data": {
+				"recordType": "ChangeSet",
+				"collectionReadEndpoint": {
+					"url": "admin\/campaigns\/sets",
+					"method": "GET"
+				},
+				"itemReadEndpoint": {
+					"url": "admin\/campaigns\/set\/:id",
+					"method": "GET"
+				},
+				"itemUpdateEndpoint": {
+					"url": "admin\/campaigns\/set\/:id",
+					"method": "PUT"
+				},
+				"itemCreateEndpoint": {
+					"url": "admin\/campaigns\/set\/:id",
+					"method": "POST"
+				},
+				"itemDeleteEndpoint": {
+					"url": "admin\/campaigns\/set\/:id",
+					"method": "DELETE"
+				},
+				"editFormSchemaEndpoint": "admin\/campaigns\/schema\/DetailEditForm",
+				"columns": [
+					{"name": "Title", "field": "Name"},
+					{"name": "Changes", "field": "_embedded.ChangeSetItems.length"},
+					{"name": "Description", "field": "Description"}
+				]
+			}
+		}, {
+			"name": "SecurityID",
+			"id": "Form_EditForm_SecurityID",
+			"type": "Hidden",
+			"component": null,
+			"holder_id": null,
+			"title": "Security ID",
+			"source": null,
+			"extraClass": "hidden",
+			"description": null,
+			"rightTitle": null,
+			"leftTitle": null,
+			"readOnly": false,
+			"disabled": false,
+			"customValidationMessage": "",
+			"attributes": [],
+			"data": []
+		}],
+		"actions": []
+	}
+}
+JSON;
+
+		$formName = $request->param('ID');
+		if($formName == 'EditForm') {
+			$response = $this->getResponse();
+			$response->addHeader('Content-Type', 'application/json');
+			$response->setBody($json);
+			return $response;
+		} else {
+			return parent::schema($request);
+		}
+	}
+
+	/**
+	 * REST endpoint to create a campaign.
+	 *
+	 * @param SS_HTTPRequest $request
+	 *
+	 * @return SS_HTTPResponse
+	 */
+	public function createCampaign(SS_HTTPRequest $request) {
+		$response = new SS_HTTPResponse();
```

Comment:
We do this pattern a lot. Worth abstracting? (`new SS_HTTPResponse("application/json", ['campaign' => 'create']);` or `new SS_JSONResponse(['campaign' => 'create']);`. Again, could wait if we expect to replace.